project_sale_order_link/models/project.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo import api, fields, models
+
+
+class Project(models.Model):
+    _inherit = "project.project"
+
+    order_line_link_ids = fields.Many2many(
+        "sale.order.line",
+        "project_project_sale_order_line_related_rel",
+        compute="_compute_order_link_ids",
+        compute_sudo=True,
+        store=True,
+    )
+    sale_order_link_ids = fields.Many2many(
+        "sale.order",
+        "project_project_sale_order_related_rel",
+        compute="_compute_order_link_ids",
+        compute_sudo=True,
+        store=True,
+    )
+
+    @api.depends(
+        "sale_line_id",
+        "task_ids.sale_line_id",
+        "timesheet_ids.so_line",
+        "sale_line_employee_ids.sale_line_id",
+    )
+    def _compute_order_link_ids(self):
+        for project in self:
+            order_lines = self.env["sale.order.line"]
+            order_lines |= (
+                project.sale_line_id
+                | project.task_ids.sale_line_id
+                | project.timesheet_ids.so_line
+                | project.sale_line_employee_ids.sale_line_id
+            )
+            project.order_line_link_ids = order_lines.ids
+            project.sale_order_link_ids = order_lines.order_id.ids
+
+    def action_view_so_link(self):
+        self.ensure_one()
+        so_ids = self.sale_order_link_ids.ids
```

Comment:
Move order lines choose behaviour to button. (Check final comments)

```suggestion
        so_ids = self.env["sale.order"]
        so_ids |= (
                project.sale_line_id.order_id
                | project.task_ids.sale_line_id.order_id
                | project.timesheet_ids.so_line.order_id
                | project.sale_line_employee_ids.sale_line_id.order_id
         ).ids
```